### ISSUE TITLE
FB Like, Twitter Follow and Google +1 buttons

### ISSUE THREAD

**Issue by quintesse** [eclipse/ceylon-lang.org#126]
Could we find a place for those somewhere? It will make it very easy for people to immediately show their appreciation (and basically they will be reminded each time they visit the page).

https://dev.twitter.com/docs/follow-button

http://www.google.com/webmasters/+1/button/

http://developers.facebook.com/docs/reference/plugins/like-box/

**Comment by quintesse** [comment]
I was thinking about some kind of side-bar that floats to the left of the page if you have enough space and collapses to one of those new-fangled side-tabs that you can click on to expand/collapse when there isn't.

**Comment by quintesse** [comment]
@FroMage isn't there someone in RedHat we could ask to do this?

**Comment by lucaswerkmeister** [comment]
By the way, please use [Social Share Privacy](https://panzi.github.io/SocialSharePrivacy/) to prevent FB et al. from tracking everyone who visits our website.

**Comment by quintesse** [comment]
I understand, but that does defeat most of the purpose of those buttons. You want the user to see how many people already liked your page, it's a psychological effect. I mean the whole purpose of those buttons is "to be social". That they are tracking that people (generic hits) are visiting our website is not something we would mind sharing, right? And at that site you link to I don't see them saying that they can track **you** specifically, or can they?

**Comment by lucaswerkmeister** [comment]
I disagree. The main purpose of these buttons, for me, is to not be there, because I’m not interested in them, and I see zero reason why Facebook should know which sites on `ceylon-lang.org` I’ve visited. (Of course they track me specifically – at least if I’m logged in, and probably IP-based if I’m not. In the case of a Like button I assume the text has to change depending on whether I already liked the link or not.) If I was actually interested in that feature, I’m sure I wouldn’t mind the extra click.

**Comment by lucaswerkmeister** [comment]
[Facebook tracks you even if you opt out](http://www.theguardian.com/technology/2015/mar/31/facebook-tracks-all-visitors-breaching-eu-law-report)

**Comment by tombentley** [comment]
>  they can track you specifically

Get real! Google, Facebook (etc etc) make money from selling ads. I fully expect them to use those buttons to follow me across sites in an effort to serve me up more relevant ads. Even if I'm not logged in. Even if I've not got an account with their service. They don't need to know who I am to know I was just visiting a site looking at xy, so serve up ads about xyz. After all, why wouldn't they do this, since it's in their financial interests to do so?

**Comment by quintesse** [comment]
Well I've got Ad Block so I don't know if they're succeeding at finding more relevant ads for me ^^

But yes, given the fact that other sites might now you're logged into FB or not it's obvious they *can*. But just like with ads I think that if you really care about that it's up to you (and partly up to the law) to make sure they can't or only when and how you want.

Those buttons are everywhere now and serve a real and useful purpose for the sites displaying them and I'd argue even for the people visiting them. If we're going to put them behind privacy buttons we might just as well get rid of them, nobody is going to click them.

**Comment by lucaswerkmeister** [comment]
> serve a real and useful purpose for the sites displaying them

Don’t we have Google Analytics for that?

**Comment by lucaswerkmeister** [comment]
For comparison, none of [Scala](http://scala-lang.org/community/index.html), [Kotlin](http://kotlinlang.org/), [Dart](https://www.dartlang.org/), [Python](https://www.python.org/), [Go](https://golang.org/), or [Rust](http://www.rust-lang.org/) have these integration buttons. They simply don’t belong on a programming language’s website IMO.

**Comment by quintesse** [comment]
Well actually several of them *do* have buttons to the major social platforms, they are not "live" buttons showing numbers or your "like status", indeed, but saying that those social buttons have no place on a programming language website is obviously false.

**Comment by lucaswerkmeister** [comment]
They have styled links to their social media pages. That’s fine, of course that belongs there. What doesn’t belong on a programming language website is “share this article on Facebook”, with a page counter and user tracker.

I thought that you wanted to add the latter kind of button. If you only want to link to our other pages, that’s fine.

**Comment by quintesse** [comment]
No, I was proposing the ones we have on the Web Runner site (which I know you won't like either), they are not "share this on facebook" buttons, but they do show *if* you +1'd, liked or followed.

**Comment by lucaswerkmeister** [comment]
You guessed right, I dislike those too :)

Simple, ordinary links to the profiles should be enough IMO.

**Comment by tombentley** [comment]
Honestly, I just don't see the point in this. Isn't the world over these stupid social media buttons now? Can we just close this?

**Comment by quintesse** [comment]
I don't agree they are "over" this. Not as long as we obviously still have a way to go to make Ceylon reach the public eye. Seems to me that every tiny bit helps.
